Assets/MixedRealityToolkit.Services/InputSimulation/Implementation/MouseRotationProvider.cs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Utility class to manage toggling of mouse rotation and associated features,
+    /// such as cursor visibility/locking
+    /// </summary>
+    /// <param name="wantsJumping">Show the cursor</param>
+    public class MouseRotationProvider
+    {
+        private bool isRotating = false;
+        public bool IsRotating => isRotating;
+
+        private static int numRotating = 0;
+        private static bool isMouseJumping = false;
+        private static bool wasCursorVisible = true;
+
+        public void Update(KeyBinding rotationKey, KeyBinding cancelRotationKey, bool toggle)
+        {
+            bool wasRotating = isRotating;
+
+            // Only allow the mouse to control rotation when Unity has focus.
+            // This enables the player to temporarily alt-tab away without having the player
+            // look around randomly back in the Unity Game window.
+            if (!Application.isFocused)
+            {
+                isRotating = false;
+            }
+            else
+            {
+                if (toggle)
+                {
+                    if (isRotating)
+                    {
+                        // pressing escape will stop capture
+                        isRotating = !KeyInputSystem.GetKeyDown(cancelRotationKey);
+                    }
+                    else
+                    {
+                        // any kind of click will capture focus
```

Comment:
>// any kind of click will capture focus [](start = 24, length = 39)

I think the comment here and the code here don't seem to match up - it looks like it's getting keydown for the rotation key specifically, not any type of click.

Reply:
Legacy comment, fixed.